libs/blocks/notification/notification.js

```diff
@@ -89,14 +89,78 @@ function wrapCopy(foreground) {
 function addCloseAction(el, btn) {
   btn.addEventListener('click', (e) => {
     if (btn.nodeName === 'A') e.preventDefault();
+
+    // Create accessibility live region
+    const liveRegion = createTag('div', {
+      class: 'notification-visibility-hidden',
+      'aria-live': 'assertive',
+      'aria-atomic': 'true',
+      role: 'status',
+      tabindex: '-1',
+    });
+    liveRegion.textContent = 'Banner closed';
+    document.body.appendChild(liveRegion);
+    liveRegion.focus();
+    let isSticky = false;
+    let rect;
+    const sectionElement = el.closest('.section');
+    if (sectionElement) {
+      sectionElement.classList.forEach((cls) => {
+        if (cls.includes('sticky')) {
+          isSticky = true;
+          rect = sectionElement.getBoundingClientRect();
+        }
+      });
+    }
     el.style.display = 'none';
     el.closest('.section')?.classList.add('close-sticky-section');
     document.dispatchEvent(new CustomEvent('milo:sticky:closed'));
+
+    setTimeout(() => {
+      const focusableSelector = 'button, [href], input, select, textarea, [tabindex]:not([tabindex="-1"])';
+      const selectedSelector = '[aria-selected="true"], [aria-checked="true"]';
+      let focusTarget;
+
+      const findFocusableInSection = (section) => {
+        if (!section) return null;
+
+        const selectedElement = section.querySelector(selectedSelector);
+        if (selectedElement) return selectedElement;
+
+        const focusableElements = [...section.querySelectorAll(focusableSelector)];
+        return focusableElements.length > 0
+          ? focusableElements[focusableElements.length - 1]
+          : null;
+      };
+
+      if (isSticky) {
+        const elementAtPosition = document.elementFromPoint(rect.left, rect.top);
+        const stickySection = !elementAtPosition.classList.contains('section') ? elementAtPosition.closest('.section') : elementAtPosition;
+        focusTarget = findFocusableInSection(stickySection);
+      }
+
+      let currentSection = el.closest('.section').previousElementSibling;
+      while (currentSection && !focusTarget) {
+        focusTarget = findFocusableInSection(currentSection);
+        if (!focusTarget) currentSection = currentSection.previousElementSibling;
+      }
+
+      if (!focusTarget) {
+        const header = document.querySelector('header');
+        if (header) {
+          const headerFocusable = [...header.querySelectorAll(focusableSelector)];
+          focusTarget = headerFocusable[headerFocusable.length - 1];
+        }
+      }
+
+      liveRegion?.remove();
+      if (focusTarget) focusTarget.focus();
```

Comment:
To prevent scrolling the element into view you can use
```suggestion
      if (focusTarget) focusTarget.focus({ preventScroll: true });
```